integration/systemd/amazon-ecs-agent.service

```diff
@@ -0,0 +1,138 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the
+# "License"). You may not use this file except in compliance
+# with the License. A copy of the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and
+# limitations under the License.
+
+# A systemd unit file to run `ecs-agent` in a Docker container. This service
+# attempts to duplicate the functionality of the `ecs-init` Golang package. The
+# notable differences currently are:
+#
+# 1. The unit file does an unconditional pull of the latest ECS Agent from
+# Docker Hub at startup.
+# 2. The unit file does not currently handle the Agent's self-upgrade
+# functionality.
+
+[Unit]
+Description=Amazon ECS Agent
+Documentation=http://docs.aws.amazon.com/AmazonECS/latest/developerguide/ECS_agent.html
+Requires=docker.service
+After=docker.service
+
+[Install]
+WantedBy=multi-user.target
+
+[Service]
+# The logic around the ECS Agent restart:
+#   - Restart on failure
+#   - Wait 15 seconds between restart
+#   - If the ECS Agent restarts more then 10 times in 5 minutes,
+# it will stay failed
+Type=simple
+Restart=on-failure
+RestartSec=5s
+RestartPreventExitStatus=5
+StartLimitInterval=5min
+StartLimitBurst=10
+
+# It is recommended in production to version pin the ECS Agent 
+# to a known version vs running the latest flag on the docker container.
+# See https://github.com/aws/amazon-ecs-agent/releases for agent versions.
+
+# It is recommended to use a systemd drop-in file vs editing this line or file.
+# Your drop-in file could look like this:
+#
+#       [Service]
+#       Environment=ECS_AGENT_VERSION=v1.17.2
+#
+# Depending on the OS, it could be placed in 
+# /etc/systemd/system/amazon-ecs-agent.d , and named 
+# ecs_agent_version_override.conf
+Environment=ECS_AGENT_VERSION=v1.17.2
+
+ExecStartPre=-/usr/bin/echo "Amazon ECS Agent systemd unit file version 1.0.0"
+
+# Load an updated ECS Agent, if it exists:
+ExecStartPre=-/bin/sh -c 'test -f /var/cache/ecs/desired-image && \
+    docker load --quiet --input=/var/cache/ecs/desired-image \
+    && rm -f $(cat /var/cache/ecs/desired-image) /var/cache/ecs/desired-image'
+
+# If we don't have an ECS Agent, load from disk, if possible, or Docker Hub:
+ExecStartPre=/bin/sh -c "`docker inspect amazon/amazon-ecs-agent &>/dev/null` \
```

Comment:
The &> redirection syntax is a bash extension. We don't want to assume that /bin/sh is bash, as this isn't the case on all systems. Please stick with POSIX-friendly redirections, so `> /dev/null 2>&1`

(Note that this issue is present in multiple places.)